data/scenarios/Challenges/Ranching/gated-paddock.yaml

```diff
@@ -0,0 +1,484 @@
+          c2 <- boolToInt b2;
+
+          turn left;
+          b3 <- isBlockedOrFenced;
+          c3 <- boolToInt b3;
+
+          turn left;
+          b4 <- isBlockedOrFenced;
+          c4 <- boolToInt b4;
+
+          return $ c1 + c2 + c3 + c4;
+          end;
+
+      // Step forward, observing left and right.
+      def observeLeftAndRight =
+          move;
+          turn left;
+          amBlockedLeft <- isBlockedOrFenced;
+          val1 <- boolToInt amBlockedLeft;
+
+          turn back;
+          amBlockedRight <- isBlockedOrFenced;
+          val2 <- boolToInt amBlockedRight;
+
+          turn right;
+          move;
+          return $ val1 + val2;
+          end;
+
+
+      /** If the four cardinal directions have at most
+      one blockage, then there will exist an orientation
+      where both that direction and its opposite direction
+      are clear.
+      So we can step that direction, check to the left and
+      right of us, then step in the opposite direction
+      and do the same. This allows us to check the 4
+      blocks that touch the corners of the center block.
+      */
+      def countDiagonalBlockages =
+          // First, orient to the clear front-to-back path
+          amBlocked <- isBlockedOrFenced;
+          if amBlocked {turn left;} {};
+
+          // Second, step to both sides
+          fwdCount <- observeLeftAndRight;
+          backCount <- observeLeftAndRight;
+          return $ fwdCount + backCount;
+          end;
+
+      def isStandingOnBridge =
+          onFence <- ishere "fence";
+          onGate <- ishere "gate";
+          if (onFence || onGate) {
+              adjCount <- countAdjacentBlockages;
+              if (adjCount > 1) {
+                  return true;
+              } {
+                  diagCount <- countDiagonalBlockages;
+                  return $ (adjCount + diagCount) > 1;
+              };
+          } {return false};
+          end;
+
+
+      /**
+      There are 3 sheep.
+      They have indices 1, 2, 3.
+      (The base has index 0).
+      */
+      def allSheep = \predicateCmd. \i.
+
+          if (i > 0) {
+              r <- robotnumbered i;
+              didSucceed <- as r {predicateCmd};
+              if didSucceed {
+                  allSheep predicateCmd $ i - 1;
+              } {
+                  return false;
+              };
+          } {
+              return true;
+          }
+          end;
+
+
+      justFilledGap <- as base {
+          isStandingOnBridge;
+      };
+
+      if (justFilledGap) {
+          allSheep checkIsEnclosed 3;
+      } {
+          return false;
+      }
+  - goal:
+      - |
+        Safe! Your sheep are now hungry.
+        Offer them something tasty and you may be rewarded.
+      - |
+        The sheep will move toward something edible on
+        an adjacent tile and will eat it if they walk over it.
+      - |
+        You may want to add a gate to the fence
+        to give yourself easier access.
```

Comment:
>  I also added two more goals and tweaked the map.

Neat! I have not saved all of them, so I will have to start over and actually finish the challenge. :open_mouth: